test/common/router/config_impl_test.cc

```diff
@@ -975,7 +975,10 @@ TEST_F(RouteMatcherTest, TestRoutes) {
     EXPECT_EQ("www2", route->clusterName());
     EXPECT_EQ("www2", virtualHostName(route));
     route->finalizeRequestHeaders(headers, stream_info, true);
+    // TODO(samflattery): remove after https://github.com/google/re2/issues/272 resolved
```

Comment:
We can add
```
#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
#undef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
#endif
```

at the top of the file :) and remove all this

Reply:
oh cool!  will that just undefine it for this file?

Reply:
I think preprocessor macros will be scoped to a single compilation unit, and it shouldn't affect a separate unit. But I wonder if it is considered the same unit as `route_fuzz_test`. Might be worth a local test via a log statement

Reply:
Sure I'll check and update you

Reply:
This doesn't work for some reason, it still fails with this change. 

Does the fuzzer set some other flags that RE2 might be using?

Reply:
Interesting, it worked for me locally! Maybe it's something with this branch, I'll pull it.

Reply:
Kk, definitely won't work until RE2 is built without the define. We'll leave this TODO as is, and I'll re-visit this later